x-pack/plugins/security_solution/common/endpoint/models/event.ts

```diff
@@ -52,3 +52,136 @@ export function parentEntityId(event: ResolverEvent): string | undefined {
   }
   return event.process.parent?.entity_id;
 }
+
+/**
+ * @param event The event to get the category for
+ */
+export function primaryEventCategory(event: ResolverEvent): string {
+  // Returning "Process" as a catch-all here because it seems pretty general
+  let eventCategoryToReturn: string = 'Process';
+  if (isLegacyEvent(event)) {
+    const legacyFullType = event.endgame.event_type_full;
+    if (legacyFullType) {
+      return legacyFullType;
+    }
+  } else {
+    const eventCategories = event.event.category;
+    const category =
+      typeof eventCategories === 'string' ? eventCategories : eventCategories[0] || '';
+
+    if (category) {
+      eventCategoryToReturn = category;
+    }
+  }
+  return eventCategoryToReturn;
+}
+
+/**
+ * ECS event type will be things like 'creation', 'deletion', 'access', etc.
+ * see: https://www.elastic.co/guide/en/ecs/current/ecs-event.html
+ * @param event The ResolverEvent to get the ecs type for
+ */
+export function ecsEventType(event: ResolverEvent): string[] {
+  if (isLegacyEvent(event)) {
+    return [event.endgame.event_subtype_full || ''];
```

Comment:
Please remove the `''` default. Returning whatever is on the event should be fine.

Reply:
https://github.com/elastic/kibana/pull/69098/commits/6d76057cacadd3285c58a967e9ea90ded25d1c6f